tensorflow/c/kernels.cc

```diff
@@ -92,6 +128,29 @@ void AddTypeConstraint(TF_KernelBuilder* kernel_builder, const char* attr_name,
 }
 #undef CASE
 
+#define CASE(type)                                               \
+  case DataTypeToEnum<type>::value: {                            \
+    kernel_builder->cc_builder->TypeConstraint<type>(attr_name); \
+    break;                                                       \
+  }
+
+void AddTypeConstraint(TF_AsyncKernelBuilder* kernel_builder,
```

Comment:
can we just move this function before the #undef CASE at line 129 so you won't need to re-define the CASE macro.

Reply:
Done.